File: app/src/main/java/com/commit451/adapterlayout/sample/AdapterFlowLayoutActivity.java
package com.commit451.adapterlayout.sample;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import com.commit451.adapterflowlayout.AdapterFlowLayout;

import java.util.ArrayList;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

public class AdapterFlowLayoutActivity extends AppCompatActivity {

    Toolbar toolbar;
    AdapterFlowLayout adapterFlowLayout;

    CheeseAdapter adapter;

    private CheeseAdapter.Listener listener = new CheeseAdapter.Listener() {
        @Override
        public void onItemClicked(Cheese cheese) {
            Toast.makeText(AdapterFlowLayoutActivity.this, cheese.name + " clicked", Toast.LENGTH_SHORT)
                    .show();
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_custom_adapter_layout);
        toolbar = findViewById(R.id.toolbar);
        adapterFlowLayout = findViewById(R.id.adapter_layout);
        toolbar.setNavigationIcon(R.drawable.ic_arrow_back_24dp);
        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onBackPressed();
            }
        });
        adapter = new CheeseAdapter(listener);
        adapterFlowLayout.setAdapter(adapter);

        findViewById(R.id.add_cheese).setOnClickListener(new View.OnClickListener() {
            @Override public void onClick(View v) {
                adapter.add(Cheeses.getRandomCheese());
            }
        });
        findViewById(R.id.remove_cheese).setOnClickListener(new View.OnClickListener() {
            @Override public void onClick(View v) {
                adapter.removeLast();
            }
        });
        findViewById(R.id.new_adapter).setOnClickListener(new View.OnClickListener() {
            @Override public void onClick(View v) {
                adapterFlowLayout.setAdapter(null);
                adapter = new CheeseAdapter(listener);
                adapterFlowLayout.setAdapter(adapter);
            }
        });
        loadCheeses();
    }

    private void loadCheeses() {
        ArrayList<Cheese> cheeses = new ArrayList<>();
        for (int i=0; i<5; i++) {
            cheeses.add(Cheeses.getRandomCheese());
        }
        adapter.setData(cheeses);
    }
}

File: app/src/main/java/com/commit451/adapterlayout/sample/AdapterFlowLayoutActivity.kt
package com.commit451.adapterlayout.sample

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.commit451.adapterflowlayout.AdapterFlowLayout
import java.util.*

class AdapterFlowLayoutActivity : AppCompatActivity() {

    private lateinit var toolbar: Toolbar
    private lateinit var adapterFlowLayout: AdapterFlowLayout

    private lateinit var adapter: CheeseAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val listener = { cheese: Cheese ->
            Toast.makeText(this, "${cheese.name} clicked", Toast.LENGTH_SHORT)
                .show()
        }
        setContentView(R.layout.activity_custom_adapter_layout)
        toolbar = findViewById(R.id.toolbar)
        adapterFlowLayout = findViewById(R.id.adapter_layout)
        toolbar.setNavigationIcon(R.drawable.ic_arrow_back_24dp)
        toolbar.setNavigationOnClickListener { onBackPressed() }
        adapter = CheeseAdapter(listener)
        adapterFlowLayout.adapter = adapter

        findViewById<View>(R.id.add_cheese).setOnClickListener { adapter.add(Cheeses.randomCheese) }
        findViewById<View>(R.id.remove_cheese).setOnClickListener { adapter.removeLast() }
        findViewById<View>(R.id.new_adapter).setOnClickListener {
            adapterFlowLayout.adapter = null
            adapter = CheeseAdapter(listener)
            adapterFlowLayout.adapter = adapter
        }
        loadCheeses()
    }

    private fun loadCheeses() {
        val cheeses = ArrayList<Cheese>()
        for (i in 0..4) {
            cheeses.add(Cheeses.randomCheese)
        }
        adapter.setData(cheeses)
    }
}
